app/cdap/components/PipelineTriggers/PipelineListTab/PipelineTriggersRow.tsx

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { useReducer } from 'react';
+import styled from 'styled-components';
+import PipelineTriggersStore from 'components/PipelineTriggers/store/PipelineTriggersStore';
+import PipelineTriggersTypes from 'components/PipelineTriggers/store/PipelineTriggersTypes';
+import PipelineTriggersActions from 'components/PipelineTriggers/store/PipelineTriggersActions';
+import { enableSchedule } from 'components/PipelineTriggers/store/PipelineTriggersActionCreator';
+import PayloadConfigModal from 'components/PipelineTriggers/PayloadConfigModal';
+import T from 'i18n-react';
+import Checkbox from '@material-ui/core/Checkbox';
+import {
+  ActionButtonsContainer,
+  CheckboxItemContainer,
+  ErrorText,
+  EventsList,
+  HelperText,
+  PipelineDescription,
+  PipelineName,
+  PipelineTriggerButton,
+  StyledAccordion,
+  StyledAccordionDetails,
+  StyledAccordionSummary,
+} from 'components/PipelineTriggers/shared.styles';
+import { IProgramStatusTrigger } from 'components/PipelineTriggers/store/ScheduleTypes';
+
+const TRIGGER_PREFIX = 'features.PipelineTriggers';
+const PREFIX = `${TRIGGER_PREFIX}.SetTriggers`;
+
+const PipelineLink = styled.a`
+  margin-left: 10px;
+`;
+
+interface IPipelineTriggersRowViewProps {
+  isExpanded: boolean;
+  onToggle: (pipeline: string) => void;
+  pipelineRow: string;
+  triggeringPipelineInfo: any;
+  triggeredPipelineInfo: any;
+  selectedNamespace: string;
+  configureError: string;
+  onPayloadToggle: (isOpen: boolean) => void;
+  payloadModalIsOpen: boolean;
+  andTriggersEnabled: boolean;
+  pipelineName: string;
+  workflowName: string;
+  triggersGroupToAdd: IProgramStatusTrigger[];
+  triggersGroupRunArgsToAdd: any;
+}
+
+const PipelineTriggersRow = ({
+  isExpanded,
+  onToggle,
+  pipelineRow,
+  triggeringPipelineInfo,
+  triggeredPipelineInfo,
+  selectedNamespace,
+  configureError,
+  onPayloadToggle,
+  payloadModalIsOpen,
+  andTriggersEnabled,
+  pipelineName,
+  workflowName,
+  triggersGroupToAdd,
+  triggersGroupRunArgsToAdd,
+}: IPipelineTriggersRowViewProps) => {
+  const initialState = {
+    completed: true,
+    killed: false,
+    failed: false,
+  };
+
+  const reducer = (oldstate, action) => {
```

Comment:
can you move this out of the component so it doesn't redefine itself every render